lisa/runners/lisa_runner.py

```diff
@@ -290,30 +290,46 @@ def _prepare_environments(self) -> None:
     def _deploy_environment_task(
         self, environment: Environment, test_results: List[TestResult]
     ) -> None:
-        try:
+        max_retries = test_results[0].runtime_data.retry_on_deployment + 1
+        for attempt in range(max_retries):
             try:
-                self.platform.deploy_environment(environment)
-                assert (
-                    environment.status == EnvironmentStatus.Deployed
-                ), f"actual: {environment.status}"
-                self._reset_awaitable_timer("deploy")
-            except ResourceAwaitableException as identifier:
-                if self._is_awaitable_timeout("deploy"):
-                    self._log.info(
-                        f"[{environment.name}] timeout on waiting for more resource: "
-                        f"{identifier}, skip assigning case."
+                try:
+                    # Attempt to deploy the environment
+                    self.platform.deploy_environment(environment)
+                    assert (
+                        environment.status == EnvironmentStatus.Deployed
+                    ), f"actual: {environment.status}"
+                    self._reset_awaitable_timer("deploy")
+                    break
+                except ResourceAwaitableException as identifier:
+                    if self._is_awaitable_timeout("deploy"):
+                        self._log.info(
+                            f"[{environment.name}] timeout on waiting for more "
+                            f"resource: {identifier}, skip assigning case."
+                        )
+                        raise SkippedException(identifier)
+                    else:
+                        # rerun prepare to calculate resource again.
+                        environment.status = EnvironmentStatus.New
+            except Exception as identifier:
```

Comment:
Use the same level as ResourceAwaitableException. The `ResourceAwaitableException` shouldn't be counted in retry.